config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -105,8 +109,18 @@
                 Method setter = beanProp.getWriteMethod();
                 Type parameterType = setter.getGenericParameterTypes()[0];
                 Class<?> parameterClass = setter.getParameterTypes()[0];
-                Object unwrapped = getValue(clazz, parameterType, parameterClass, config, originalNames.get(beanProp.getName()));
-                setter.invoke(bean, unwrapped);
+                String configPropName = originalNames.get(beanProp.getName());
+                if (configPropName == null) {
+                    configPropName = beanProp.getName();
+                }
```

Comment:
The `beanProp.getName()` will not appear in `originNames` if the `config` (file) does not contain an entry for that property to begin with. Added a test case called `testNotAnOptionalProperty` to cover this codepath.